### PR TITLE
MainClassBuildStep - log error when app fails to start

### PR DESCRIPTION
**Pull request by mkouba**
- remove Throwable.printStackTrace()

### CODE REVIEW

**Review by stuartwdouglas — APPROVED** [review]

File: core/deployment/src/main/java/io/quarkus/deployment/steps/MainClassBuildStep.java [hunk 1]
```diff
@@ -235,9 +244,11 @@ MainClassBuildItem build(List<StaticBytecodeRecorderBuildItem> staticInitTasks,
                 tryBlock.load(LaunchMode.DEVELOPMENT.equals(launchMode.getLaunchMode())));
 
         cb = tryBlock.addCatch(Throwable.class);
-        cb.invokeVirtualMethod(ofMethod(Throwable.class, "printStackTrace", void.class), cb.getCaughtException());
+        cb.invokeVirtualMethod(ofMethod(Logger.class, "error", void.class, Object.class, Throwable.class),
+                cb.readStaticField(logField.getFieldDescriptor()), cb.load("Failed to start application"),
+                cb.getCaughtException());
         cb.invokeVirtualMethod(ofMethod(StartupContext.class, "close", void.class), startupContext);
-        cb.throwException(RuntimeException.class, "Failed to start quarkus", cb.getCaughtException());
+        cb.throwException(RuntimeException.class, "Failed to start quarkus");
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/steps/MainClassBuildStep.java]: There are test failures related to this change.

From what I saw in the log, the exception wasn't logged twice so I would keep the cause here.

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks @mkouba . It wasn't an Arc issue after all then. Could you have a look at my comment?

File: core/deployment/src/main/java/io/quarkus/deployment/steps/MainClassBuildStep.java (hunk 1 above)
Comment (mkouba) [reply core/deployment/src/main/java/io/quarkus/deployment/steps/MainClassBuildStep.java]: Hm, I tested my PR with quickstart and it was logged twice when I tried to run the app using the `-runner.jar`. I can try the dev mode and native image too..

Comment (mkouba) [reply core/deployment/src/main/java/io/quarkus/deployment/steps/MainClassBuildStep.java]: Actually some ArC tests failed because of this ;-).

Comment (mkouba) [reply core/deployment/src/main/java/io/quarkus/deployment/steps/MainClassBuildStep.java]: So I verified that it's logged twice but I agree that we should keep the cause here... I will update the PR shortly.

**Comment by mkouba** [comment]
The windows job failed with:
```
Caused by: org.quartz.JobPersistenceException: Couldn't check for existence of job: Table "QRTZ_JOB_DETAILS" not found; SQL statement:
SELECT JOB_NAME FROM QRTZ_JOB_DETAILS WHERE SCHED_NAME = 'QuarkusQuartzScheduler' AND JOB_NAME = ? AND JOB_GROUP = ? [42102-197] [See nested exception: org.h2.jdbc.JdbcSQLException: Table "QRTZ_JOB_DETAILS" not found; SQL statement:
SELECT JOB_NAME FROM QRTZ_JOB_DETAILS WHERE SCHED_NAME = 'QuarkusQuartzScheduler' AND JOB_NAME = ? AND JOB_GROUP = ? [42102-197]
```

**Comment by gsmet** [comment]
@mkouba so I might be saying something stupid but if it's logged twice, should we simply get rid of it?

**Comment by mkouba** [comment]
@gsmet the problem is that afterwards we will not be able to assert the build exception in tests and possibly in the dev mode?

**Review by gsmet — APPROVED** [review]

**Pull request merged**